core/provider/service.go

```diff
@@ -195,12 +215,19 @@ func (s *Service) Update(ctx context.Context, p *domain.Provider) error {
 		}
 	}
 
+	if p.Config.Credentials != nil {
+		// decode any base64 encoded string within credentials
+		decodeBase64Values(p.Config.Credentials)
+	}
```

Comment:
Sorry I kinda forgot with this but is there any reason to do base64 decoding inside this business logic instead of doing it right after reading it from Config file?